deepsource/client.go

```diff
@@ -35,14 +35,24 @@ func (c Client) GetToken() string {
 
 // Returns a new GQLClient
 func New() (*Client, error) {
-	// TODO: Figure out better way to resolve host here
-	apiClientURL := fmt.Sprintf("https://api.%s/graphql/", config.Cfg.Host)
+	apiClientURL := getAPIClientURL()
 	gql := graphql.NewClient(apiClientURL)
 	return &Client{
 		gql: gql,
 	}, nil
 }
 
+// Formats and returns the DeepSource Public API client URL
+func getAPIClientURL() string {
+	apiClientURL := fmt.Sprintf("https://api.%s/graphql/", config.Cfg.Host)
+
+	// Check if the domain is different from the default domain (In case of Enterprise users)
+	if config.Cfg.Host != config.DefaultHostName {
+		apiClientURL = fmt.Sprintf("https://%s/api/graphql/", config.Cfg.Host)
+	}
+	return apiClientURL
+}
+
```

Comment:
We should ideally not have any dependencies for this package.  The dependency should always point inwards.

Reply:
Have used a struct `ClientProperties` which the function that requires a client can send as param and get the corresponding client. This way, the SDK won't fetch it directly from the config package.

Reply:
The client can be initialized once.  And the New() function in Client should accept the Opts param.

Reply:
Yes, correct. The `New()` function receives these params.